### DISCUSSION TITLE
Would be great if grub-btrfs can also integrate snap-pac

### DISCUSSION THREAD

**Discussion by RedBlizard** [Antynea/grub-btrfs#302] [Ideas]
Hi guys i think it would be really great if grub-btrfs can also integrate snap-pac.
Dont know if this is possible but its just a idea .

Here are some of the benefits of integrating the snap-pac hook options into grub-btrfs:

It would provide a more seamless way to use snap-pac with grub-btrfs.
It would allow users to use the full range of snap-pac hook options with grub-btrfs.
It would make it easier for users to troubleshoot problems with snap-pac and grub-btrfs.
Here are some of the challenges of integrating the snap-pac hook options into grub-btrfs:

It would require some work to modify the grub-btrfs source code or to add a new configuration file.
It would need to be carefully tested to ensure that it does not break anything else in grub-btrfs.
It would need to be documented so that users know how to use it.

**Comment by Schievel1** [comment]
I don’t really see the benefit. Why would it provide a more seamless way? Snap-pac triggers when you install a package. Grub-btrfs runs when grub mkconfig runs. You just need a way to trigger grub-mkconfig from
Snap-pac, that’s all. It doesn’t get any more seamless than that. 

> It would allow users to use the full range of snap-pac hook options with grub-btrfs.

how is that? Are there snap-pac hook options right now that users can’t use with grub-btrfs?

> It would make it easier for users to troubleshoot problems with snap-pac and grub-btrfs.

no it wouldn’t. In fact it would make it harder. Now, if you have a problem with snapshots not being triggered when a package is installed, you know it has something to do with snap-pac. (Or whatever you’re using)
If the snapshots don’t show up in the grub menu the problem is grub-btrfs. If you integrate both the Problem could be anywhere. 

Another problem is that pacman hooks are, well, for pacman obviously. I would prefer to stay distribution agnostic. 

Also grub-btrfs stays with the Unix Philosophy of doing one thing and do it well. You can exchange the components to your liking. Grub-btrfs doesn’t care who creates the snapshot, be it timeshift, snapped, snap-pac or even manually.
